apollo-router/src/plugins/telemetry/config_new/logging.rs

```diff
@@ -0,0 +1,68 @@
+use schemars::JsonSchema;
+use serde::Deserialize;
+
+/// Logging configuration.
+#[allow(dead_code)]
+#[derive(Deserialize, JsonSchema, Clone, Default, Debug)]
+#[serde(deny_unknown_fields, default)]
+pub(crate) struct Logging {
+    /// Settings for logging to stdout.
+    stdout: StdOut,
+    /// Settings for logging to a file.
+    file: File,
+}
+
+#[allow(dead_code)]
+#[derive(Deserialize, JsonSchema, Clone, Default, Debug)]
+#[serde(deny_unknown_fields, default)]
+struct StdOut {
+    /// Set to true to log to stdout.
+    enabled: bool,
+    /// The format to log to stdout.
+    format: Format,
+}
+
+/// Log to a file
+#[allow(dead_code)]
+#[derive(Deserialize, JsonSchema, Clone, Default, Debug)]
+#[serde(deny_unknown_fields, default)]
+struct File {
+    /// Set to true to log to a file.
+    enabled: bool,
+    /// The path pattern of the file to log to.
+    path: String,
+    /// The format of the log file.
+    format: Format,
+    /// The period to rollover the log file.
+    rollover: Rollover,
+}
+
+/// The format for logging.
+#[allow(dead_code)]
+#[derive(Deserialize, JsonSchema, Clone, Default, Debug)]
+#[serde(deny_unknown_fields, rename_all = "snake_case")]
+enum Format {
+    Aws,
+    Bunyan,
+    Gelf,
+    Google,
```

Comment:
Is it really called the `google` format ?

Reply:
The documentation is here: https://cloud.google.com/logging/docs/structured-logging
I'm not sure if it's officially called google. Let's at least add a rustdoc that details it.

Reply:
[9ac775a](https://github.com/apollographql/router/pull/4061/commits/9ac775a02e509a68dec99653ce010d109eab098e)